network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -252,4 +338,118 @@ public String toString() {
         .toString();
     }
   }
+
+  static ObjectMapper mapper = new ObjectMapper();
+
+  private static byte[] dbAppExecKey(AppExecId appExecId) throws IOException {
+    // we stick a common prefix on all the keys so we can find them in the DB
+    String appExecJson = mapper.writeValueAsString(appExecId);
+    String key = (APP_KEY_PREFIX + ";" + appExecJson);
```

Comment:
Is the prefix needed? It doesn't seem like you're using it for anything (since you just discard it in the parse method).